### DISCUSSION TITLE
Redirect traffic to url with www and https

### DISCUSSION THREAD

**Discussion by weblabjovan** [vercel/next.js#17365] [Help] [answered]
Is there a simple and clean way to ensure that all the traffic we are getting on our server that is running next.js app will land on https://www.somepage.com ?

If we have a static web page but we need to stream line the traffic to URL with https and www we need to use window objetct in our component to get the URI, than parse that URL and see if it is right or not. If it is, than nothing happens if it's not than to we redirect to a page also using window object.

> import Header from '../components/header

> const Home = () => (
> if (typeof window !== 'undefined') {
>     //isURLFormatOk is a function that parses the link  and descides if the URL is right or not, returnstrue or false
>     if (!isURLFormatOk(window.location.href)) {
>       window.location.href = 'https://www.somepage.com';
>     }
>   }
>   <>
>     <Header />
>     <h1>Hello World!</h1>
>   </>
> )
> export default  Home

If we do not want to use the window object than we have to make a dynamic page and use getInitialProps and asPath prop.

> import Header from '../components/header
> 
> const Home = () => (
>   <>
>     <Header />
>     <h1>Hello World!</h1>
>   </>
> )
> 
> Home.getInitialProps = async (context: any) => {
>   if (context['req']) {
>       //isURLFormatOk is a function that parses the link  and descides if the URL is right or not, returnstrue or false
>       if (!isURLFormatOk(context['req']['headers']['host'])) {
>        context.res.writeHead(302, {Location: 'https://www.somepage.com'});
>        context.res.end();
>       }
>     }
>   }
> export default  Home

Is there any simpler way of doing this? Maybe if there was some setting in the next.config.  Any ideas?
It would be great if we can have static page and achieve this single URL streamline without using window object.

**Comment by leerob** [comment, marked as answer]
Typically, this is handled by the hosting platform. For example, you can direct `www.site.com` to `site.com` in the Vercel project dashboard (and similar to other hosting platforms).

**Reply by weblabjovan** [reply]
This is the best solution.

I have Vercel account and I settup the redirection on my dashboard. So now I have a static page, and my traffic is streamlined on a single web source (with https and www), and I do not have to handle this in my code. Great.

Thanks Lee
